unchanged:
--- xbmc/tools/depends/target/ffmpeg/CMakeLists.txt	2023-01-29 04:12:26.179949543 +0100
+++ xbmc/tools/depends/target/ffmpeg/CMakeLists.new	2023-01-29 04:10:14.270603769 +0100
@@ -150,6 +150,10 @@
                     CONFIGURE_COMMAND ${pkgconf} ${pkgconf_path} <SOURCE_DIR>/configure
                       --prefix=${CMAKE_INSTALL_PREFIX}
                       --extra-version="kodi-${FFMPEG_VER}"
+                      --enable-version3
+                      --enable-rkmpp
+                      --enable-libdrm
+                      --disable-vulkan 
                       ${ffmpeg_conf}
                     BUILD_COMMAND ${MAKE_COMMAND})
