From af5df41ea8ec5e4603416b0e0839bff0190ce3f1 Mon Sep 17 00:00:00 2001
From: boogie <boogiepop@gmx.com>
Date: Thu, 16 Nov 2023 19:28:45 +0100
Subject: [PATCH] fix build

---
 src/egl/main/egldriver.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/egl/main/egldriver.h b/src/egl/main/egldriver.h
index 6c8ea711d7c..51c111eb0ce 100644
--- a/src/egl/main/egldriver.h
+++ b/src/egl/main/egldriver.h
@@ -227,7 +227,7 @@ struct _egl_driver {
    /* for EGL_EXT_surface_compression */
    EGLBoolean (*QuerySupportedCompressionRatesEXT)(_EGLDisplay *disp,
                                                    _EGLConfig *config,
-                                                   const EGLint *attr_list,
+                                                   const EGLAttrib *attr_list,
                                                    EGLint *rates,
                                                    EGLint rate_size,
                                                    EGLint *num_rates);
-- 
2.42.1

