diff --git a/mpp/hal/rkdec/h265d/hal_h265d_com.c b/mpp/hal/rkdec/h265d/hal_h265d_com.c
index d7e8db7..40639d4 100644
--- a/mpp/hal/rkdec/h265d/hal_h265d_com.c
+++ b/mpp/hal/rkdec/h265d/hal_h265d_com.c
@@ -88,7 +88,7 @@
 
 RK_U32 hevc_hor_align(RK_U32 val)
 {
-    if (compat_ext_non_fbc_no_256_odd)
+    if (*compat_ext_non_fbc_no_256_odd)
         return MPP_ALIGN(val, 8);
 
     return MPP_ALIGN(val, 256) | 256;
diff --git a/mpp/hal/rkdec/vp9d/hal_vp9d_com.c b/mpp/hal/rkdec/vp9d/hal_vp9d_com.c
index eacd11c..59d217b 100644
--- a/mpp/hal/rkdec/vp9d/hal_vp9d_com.c
+++ b/mpp/hal/rkdec/vp9d/hal_vp9d_com.c
@@ -997,7 +997,7 @@
 
 RK_U32 vp9_hor_align(RK_U32 val)
 {
-    if (compat_ext_non_fbc_no_256_odd)
+    if (*compat_ext_non_fbc_no_256_odd)
         return MPP_ALIGN(val, 64);
 
     return MPP_ALIGN(val, 256) | 256;
